docs/bigquery_snippets.py

```diff
@@ -0,0 +1,198 @@
+    ORIGINAL_DESCRIPTION = 'Original description'
+    PATCHED_DESCRIPTION = 'Patched description'
+    dataset = client.dataset(DATASET_NAME)
+    dataset.description = ORIGINAL_DESCRIPTION
+    dataset.create()
+    to_delete.append(dataset)
+
+    # [START dataset_patch]
+    ONE_DAY_MS = 24 * 60 * 60 * 1000
+    assert dataset.description == ORIGINAL_DESCRIPTION
+    dataset.patch(
+        description=PATCHED_DESCRIPTION,
+        default_table_expiration_ms=ONE_DAY_MS
+    )      # API request
+    assert dataset.description == PATCHED_DESCRIPTION
+    assert dataset.default_table_expiration_ms == ONE_DAY_MS
+    # [END dataset_patch]
+
+
+@snippet
+def dataset_update(client, to_delete):
+    """Update a dataset's metadata."""
+    DATASET_NAME = 'dataset_update_%d' % (_millis(),)
+    ORIGINAL_DESCRIPTION = 'Original description'
+    UPDATED_DESCRIPTION = 'Updated description'
+    dataset = client.dataset(DATASET_NAME)
+    dataset.description = ORIGINAL_DESCRIPTION
+    dataset.create()
+    to_delete.append(dataset)
+    dataset.reload()
+
+    # [START dataset_update]
+    from gcloud.bigquery import AccessGrant
+    assert dataset.description == ORIGINAL_DESCRIPTION
+    assert dataset.default_table_expiration_ms is None
+    grant = AccessGrant(
+        role='READER', entity_type='domain', entity_id='example.com')
+    assert grant not in dataset.access_grants
+    ONE_DAY_MS = 24 * 60 * 60 * 1000
+    dataset.description = UPDATED_DESCRIPTION
+    dataset.default_table_expiration_ms = ONE_DAY_MS
+    grants = list(dataset.access_grants)
+    grants.append(grant)
+    dataset.access_grants = grants
+    dataset.update()              # API request
+    assert dataset.description == UPDATED_DESCRIPTION
+    assert dataset.default_table_expiration_ms == ONE_DAY_MS
+    assert grant in dataset.access_grants
+    # [END dataset_update]
+
+
+@snippet
+def dataset_delete(client, to_delete):  # pylint: disable=unused-argument
+    """Delete a dataset."""
+    DATASET_NAME = 'dataset_delete_%d' % (_millis(),)
+    dataset = client.dataset(DATASET_NAME)
+    dataset.create()
+
+    # [START dataset_delete]
+    assert dataset.exists()       # API request
+    dataset.delete()
+    assert not dataset.exists()   # API request
+    # [END dataset_delete]
+
+
+def _find_examples():
+    funcs = [obj for obj in globals().values()
+             if getattr(obj, '_snippet', False)]
+    for func in sorted(funcs, key=lambda f: f.func_code.co_firstlineno):
+        yield func
+
+
+def main():
+    client = Client()
+    for example in _find_examples():
+        to_delete = []
+        print('%-25s: %s' % (
+            example.func_name, example.func_doc))
+        try:
+            example(client, to_delete)
+        except AssertionError as e:
+            print('   FAIL: %s' % (e,))
+        except Exception as e:  # pylint: disable=broad-except
+            print('  ERROR: %r' % (e,))
+        for item in to_delete:
+            item.delete()
```

Comment:
As with the previous round of snippets, I would much prefer we use an existing test harness instead of rolling our own. Why not use `unittest.TestCase` as we do in system tests?